src/openrct2/world/Map.cpp

```diff
@@ -1539,6 +1548,104 @@ void map_update_tiles()
     }
 }
 
+void MapInvalidateHeightCache()
+{
+    _mapHighestTileHeightLoopPosition = 0;
+}
+
+void MapUpdateHeightCache(bool fullCheck)
+{
+    constexpr int32_t mapTileCount = MAXIMUM_MAP_SIZE_TECHNICAL * MAXIMUM_MAP_SIZE_TECHNICAL;
+
+    if (fullCheck)
+        _mapHighestTileHeightLoopPosition = 0;
+
+    if (_mapHighestTileHeightLoopPosition == mapTileCount)
+        return;
+
+    // If we're doing a full check of the map then zero out the loop
+    // position to start at tile zero, checking all tiles before returning.
+    // Otherwise check a limited number of tiles per each time the
+    // function is called.
+    int32_t tilesToCheck = fullCheck ? mapTileCount : mapTileCount / 64;
```

Comment:
```suggestion
    int32_t tilesToCheck = fullCheck ? mapTileCount : mapTileCount / 8;
```
I've done some testing and 64 seems to be too large, the update takes way too long. The effect is first apparent after 2,5 seconds.

8 seems to be a good value which would mean also 8 * 40ms to update.
```
INFO[F:\C++\OpenRCT2\src\openrct2\world\Map.cpp:1626 (MapUpdateHeightCache)]: MapUpdateHeightCache ran in 0.00004600 secs
INFO[F:\C++\OpenRCT2\src\openrct2\world\Map.cpp:1626 (MapUpdateHeightCache)]: MapUpdateHeightCache ran in 0.00005200 secs
INFO[F:\C++\OpenRCT2\src\openrct2\world\Map.cpp:1626 (MapUpdateHeightCache)]: MapUpdateHeightCache ran in 0.00004600 secs
INFO[F:\C++\OpenRCT2\src\openrct2\world\Map.cpp:1626 (MapUpdateHeightCache)]: MapUpdateHeightCache ran in 0.00004400 secs
INFO[F:\C++\OpenRCT2\src\openrct2\world\Map.cpp:1626 (MapUpdateHeightCache)]: MapUpdateHeightCache ran in 0.00004200 secs
INFO[F:\C++\OpenRCT2\src\openrct2\world\Map.cpp:1626 (MapUpdateHeightCache)]: MapUpdateHeightCache ran in 0.00003700 secs
INFO[F:\C++\OpenRCT2\src\openrct2\world\Map.cpp:1626 (MapUpdateHeightCache)]: MapUpdateHeightCache ran in 0.00002800 secs
```
Its fast enough not to worry about performance on my end. But the map size may or may not increase in the future.

Reply:
Otherwise this seems to be working well, after this change its ready to be merged as far I'm concerned.

Reply:
Sounds good. Thanks for giving it a good kick of the tires. I'll make the change this evening :)